### PR TITLE
samples: matter: Added support for multi-image DFU on nRF5340DK

### PR DESCRIPTION
**Pull request by kkasperczyk-no**
Currently lock and light_bulb samples support only application
core firmware upgrade on nRF5340DK. This change enables also
upgrading network core image.

Signed-off-by: Kamil Kasperczyk <kamil.kasperczyk@nordicsemi.no>

### CODE REVIEW

**Comment by kkasperczyk-no** [comment]
Added DNM label until Matter upstream documentation change regarding performing DFU guide will be merged and pulled to the sdk-nrf.

File: samples/matter/lock/README.rst [hunk 1]
```diff
@@ -72,6 +72,28 @@ Configuration
...
+
++------------------+--------------------------------------+------------------------+------------------------------+
+| Config suffix    | Enabled feature                      | Enabling build option  | Supported boards             |
++==================+======================================+========================+==============================+
+| none             | none (basic build)                   | none                   | ``nrf52840dk_nrf52840``      |
+|                  |                                      |                        | ``nrf5340dk_nrf5340_cpuapp`` |
+|                  |                                      |                        | ``nrf21540dk_nrf52840``      |
++------------------+--------------------------------------+------------------------+------------------------------+
+| _dfu             | Single-image Device Firmware Upgrade | ``-DBUILD_WITH_DFU=1`` | ``nrf52840dk_nrf52840``      |
+|                  |                                      |                        | ``nrf21540dk_nrf52840``      |
```
Comment (LuDuda) [inline samples/matter/lock/README.rst:90]: This is not supported.

Comment (kkasperczyk-no) [reply samples/matter/lock/README.rst:90]: Right, I suggested by looking at nrf21540dk_nrf52840_dfu.conf file name. But I think this file is not used anywhere, so I will remove it.

File: samples/matter/lock/README.rst [hunk 2]
```diff
@@ -72,6 +72,26 @@ Configuration
...
++-------------------+--------------------------------------+------------------------+------------------------------+
+| Config suffix     | Enabled feature                      | Enabling build option  | Supported boards             |
++===================+======================================+========================+==============================+
+| none              | none (basic build)                   | none                   | ``nrf52840dk_nrf52840``      |
+|                   |                                      |                        | ``nrf5340dk_nrf5340_cpuapp`` |
++-------------------+--------------------------------------+------------------------+------------------------------+
+| _single_image_dfu | Single-image Device Firmware Upgrade | ``-DBUILD_WITH_DFU=1`` | ``nrf52840dk_nrf52840``      |
++-------------------+--------------------------------------+------------------------+------------------------------+
+| _multi_image_dfu  | Multi-image Device Firmware Upgrade  | ``-DBUILD_WITH_DFU=1`` | ``nrf5340dk_nrf5340_cpuapp`` |
++-------------------+--------------------------------------+------------------------+------------------------------+
```
Comment (wiba-nordic) [inline samples/matter/lock/README.rst:91]:
```suggestion
+---------------------------+--------------------------------------+------------------------+------------------------------+
| Config suffix             | Enabled feature                      | Enabling build option  | Supported boards             |
+===========================+======================================+========================+==============================+
| none                      | none (basic build)                   | none                   | ``nrf52840dk_nrf52840``      |
|                           |                                      |                        | ``nrf5340dk_nrf5340_cpuapp`` |
+---------------------------+--------------------------------------+------------------------+------------------------------+
| :file:`_single_image_dfu` | Single-image Device Firmware Upgrade | ``-DBUILD_WITH_DFU=1`` | ``nrf52840dk_nrf52840``      |
+---------------------------+--------------------------------------+------------------------+------------------------------+
| :file:`_multi_image_dfu`  | Multi-image Device Firmware Upgrade  | ``-DBUILD_WITH_DFU=1`` | ``nrf5340dk_nrf5340_cpuapp`` |
+---------------------------+--------------------------------------+------------------------+------------------------------+
```

**Review by wiba-nordic — CHANGES_REQUESTED** [review]
A couple of small improvements needed.

**Review by wiba-nordic — APPROVED** [review]
Looks good!

**Review by Damian-Nordic — APPROVED** [review]

**Review by LuDuda — APPROVED** [review]

**Pull request merged**